src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+#### Good
+
+```
+components/
+|- TheHeading.vue
+|- TheSidebar.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Tightly coupled component names <sup data-p="b">strongly recommended</sup>
+
+**Child components that are tightly coupled with their parent should include the parent component name as a prefix.**
+
+If a component only makes sense in the context of a single parent component, that relationship should be evident in its name. Since editors typically organize files alphabetically, this also keeps these related files next to each other.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+```
+components/
+|- TodoList.vue
+|- TodoItem.vue
+|- TodoButton.vue
+```
+
+```
+components/
+|- SearchSidebar.vue
+|- NavigationForSearchSidebar.vue
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- TodoList.vue
+|- TodoListItem.vue
+|- TodoListItemButton.vue
+```
+
+```
+components/
+|- SearchSidebar.vue
+|- SearchSidebarNavigation.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Order of words in component names <sup data-p="b">strongly recommended</sup>
+
+**Component names should start with the highest-level (often most general) words and end with descriptive modifying words.**
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+You may be wondering:
+
+> "Why would we force component names to use less natural language?"
+
+In natural English, adjectives and other descriptors do typically appear before the nouns, while exceptions require connector words. For example:
+
+- Coffee _with_ milk
+- Soup _of the_ day
+- Visitor _to the_ museum
+
+You can definitely include these connector words in component names if you'd like, but the order is still important.
+
+Also note that **what's considered "highest-level" will be contextual to your app**. For example, imagine an app with a search form. It may include components like this one:
+
+```
+components/
+|- ClearSearchButton.vue
+|- ExcludeFromSearchInput.vue
+|- LaunchOnStartupCheckbox.vue
+|- RunSearchButton.vue
+|- SearchInput.vue
+|- TermsCheckbox.vue
+```
+
+As you might notice, it's quite difficult to see which components are specific to the search. Now let's rename the components according to the rule:
+
+```
+components/
+|- SearchButtonClear.vue
+|- SearchButtonRun.vue
+|- SearchInputExcludeGlob.vue
+|- SearchInputQuery.vue
+|- SettingsCheckboxLaunchOnStartup.vue
+|- SettingsCheckboxTerms.vue
+```
+
+Since editors typically organize files alphabetically, all the important relationships between components are now evident at a glance.
+
+You might be tempted to solve this problem differently, nesting all the search components under a "search" directory, then all the settings components under a "settings" directory. We only recommend this approach in very large apps of 100+ components, for these reasons:
```

Comment:
IMO the number `100+` is pretty arbitrary. How about just mention "apps with a large number of components" instead?

Reply:
It _is_ somewhat arbitrary, in the sense that we're using `100` rather than `99` or `101`, but I worry that if we say "apps with a large number of components" people will just ask, "How many is a large number?" 😄   

I've worked very comfortably with well over a 100 well-named components in a single directory, so I figured this was relatively low, safe number for people to start considering breaking things out. What do you think?

Reply:
> "How many is a large number?" 😄

"But that's the whole point – we've no idea! An app can be pretty complex with 50-ish components, when another with 150 components can still be considered simple enough. It depends on what you're building, your skills, your team size etc. In other words, _you_ know when to organize components into directories."

Reply:
I guess we can rewrite this in a way to emphasize that this is subjective and user-dependent?

Reply:
I guess part of my problem here is that I personally see nested component directories as an anti-pattern at virtually _every_ size, even though many others would disagree. The problem is people reach for folders to help manage their complexity instead of giving their components better names, which has many more benefits. So I worry about giving apps that are too small an excuse to do this.

Reply:
OK, I just made a small change here from:

> We only recommend this approach in very large apps of 100+ components [...]

to:

> We only recommend considering this approach in very large apps (e.g. 100+ components) [...]

Gives people a little more leeway. 🙂